src/main/java/de/zalando/tip/zalenium/servlet/LiveNodeHtmlRenderer.java

```diff
@@ -0,0 +1,226 @@
+import org.openqa.grid.web.servlet.beta.SlotsLines;
+import org.openqa.selenium.Platform;
+import org.openqa.selenium.remote.CapabilityType;
+
+import java.util.Map;
+
+public class LiveNodeHtmlRenderer implements HtmlRenderer {
+    private RemoteProxy proxy;
+
+    public LiveNodeHtmlRenderer(RemoteProxy proxy) {
+        this.proxy = proxy;
+    }
+
+    public String renderSummary() {
+        StringBuilder builder = new StringBuilder();
+        builder.append("<div class='proxy'>");
+        builder.append("<p class='proxyname'>");
+        builder.append(proxy.getClass().getSimpleName());
+
+        builder.append(getHtmlNodeVersion());
+
+        String platform = getPlatform(proxy);
+
+        builder.append("<p class='proxyid'>id : ");
+        builder.append(proxy.getId());
+        builder.append(", OS : ").append(platform).append("</p>");
+
+        builder.append(nodeTabs());
+
+        builder.append("<div class='content'>");
+
+        builder.append(tabBrowsers());
+        builder.append(tabConfig());
+
+        builder.append("</div>");
+        builder.append("</div>");
+
+        return builder.toString();
+    }
+
+    private String getHtmlNodeVersion() {
+        try {
+            JsonObject object = proxy.getStatus();
+            String version = object.get("value").getAsJsonObject()
+                    .get("build").getAsJsonObject()
+                    .get("version").getAsString();
+            return " (version : "+version+ ")";
+        }catch (Exception e) {
+            return " unknown version,"+e.getMessage();
+        }
+    }
+
+    // content of the config tab.
+    private String tabConfig() {
+        StringBuilder builder = new StringBuilder();
+        builder.append("<div type='config' class='content_detail'>");
+        Map<String, Object> config = proxy.getConfig();
+
+        for (String key : config.keySet()) {
+            builder.append("<p>");
+            builder.append(key);
+            builder.append(":");
+            builder.append(config.get(key));
+            builder.append("</p>");
+        }
+
+        builder.append("</div>");
+        return builder.toString();
+    }
+
+
+    // content of the browsers tab
+    private String tabBrowsers() {
+        StringBuilder builder = new StringBuilder();
+        builder.append("<div type='browsers' class='content_detail'>");
+
+        SlotsLines rcLines = new SlotsLines();
+        SlotsLines wdLines = new SlotsLines();
+
+        for (TestSlot slot : proxy.getTestSlots()) {
+            if (slot.getProtocol() == SeleniumProtocol.Selenium) {
+                rcLines.add(slot);
+            } else {
+                wdLines.add(slot);
+            }
+        }
+
+        if (rcLines.getLinesType().size() != 0) {
+            builder.append("<p class='protocol' >Remote Control (legacy)</p>");
+            builder.append(getLines(rcLines));
+        }
+        if (wdLines.getLinesType().size() != 0) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1155)